adaptive/learner/average_learner1D.py

```diff
@@ -0,0 +1,435 @@
+
+            # We also need to update scale and losses
+            super()._update_scale(x, y)
+            self._update_losses_resampling(x, real=True)
+
+            # If the scale has increased enough, recompute all losses.
+            # We only update the scale considering resampled points, since new
+            # points are more likely to be outliers.
+            if self._scale[1] > self._recompute_losses_factor * self._oldscale[1]:
+                for interval in reversed(self.losses):
+                    self._update_interpolated_loss_in_interval(*interval)
+                self._oldscale = deepcopy(self._scale)
+
+    def _update_distances(self, x):
+        x_left, x_right = self.neighbors[x]
+        y = self.data[x]
+        if x_left is not None:
+            self._distances[x_left] = hypot((x - x_left), (y - self.data[x_left]))
+        if x_right is not None:
+            self._distances[x] = hypot((x_right - x), (self.data[x_right] - y))
+
+    def _update_losses_resampling(self, x, real=True):
+        """Update all losses that depend on x, whenever the new point is a re-sampled point."""
+        # (x_left, x_right) are the "real" neighbors of 'x'.
+        x_left, x_right = self._find_neighbors(x, self.neighbors)
+        # (a, b) are the neighbors of the combined interpolated
+        # and "real" intervals.
+        a, b = self._find_neighbors(x, self.neighbors_combined)
+
+        if real:
+            for ival in _get_intervals(x, self.neighbors, self.nth_neighbors):
+                self._update_interpolated_loss_in_interval(*ival)
+        elif x_left is not None and x_right is not None:
+            # 'x' happens to be in between two real points,
+            # so we can interpolate the losses.
+            dx = x_right - x_left
+            loss = self.losses[x_left, x_right]
+            self.losses_combined[a, x] = (x - a) * loss / dx
+            self.losses_combined[x, b] = (b - x) * loss / dx
+
+        # (no real point left of x) or (no real point right of a)
+        left_loss_is_unknown = (x_left is None) or (not real and x_right is None)
+        if (a is not None) and left_loss_is_unknown:
+            self.losses_combined[a, x] = float("inf")
+
+        # (no real point right of x) or (no real point left of b)
+        right_loss_is_unknown = (x_right is None) or (not real and x_left is None)
+        if (b is not None) and right_loss_is_unknown:
+            self.losses_combined[x, b] = float("inf")
+
+    def _calc_error_in_mean(self, ys, y_avg, n):
+        variance_in_mean = sum((y - y_avg) ** 2 for y in ys) / (n - 1)
+        t_student = scipy.stats.t.ppf(1 - self.alpha, df=n - 1)
+        return t_student * (variance_in_mean / n) ** 0.5
+
+    def tell_many(self, xs, ys):
+        """Tell the learner about some values.
+
+        Parameters
+        ----------
+        xs : Iterable of values from the function domain
+        ys : Iterable of lists of values (several data samples)
```

Comment:
I think this breaks the API: runner isn't supposed to sort the points into lists with the same x-coordinate.

`tell` and `tell_many` must satisfy the relationship provided by the [BaseLearner](https://github.com/python-adaptive/adaptive/blob/master/adaptive/learner/base_learner.py#L87-L106). The reason why you may implement both is exclusively computational efficiency.

Reply:
I redesigned the `tell_many()` in the [last commit](https://github.com/python-adaptive/adaptive/pull/283/commits/a8c86ccae75e0f5d4da721bd1792f8c9a10c95e8), so that it matches the definition from the BaseLearner. I think now it works as it is supposed to.